pkg/filters/args.go

```diff
@@ -55,10 +58,21 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 		if !found {
 			return false
 		}
-		// TODO: use type assertion instead of string conversion
-		if argName != "syscall" {
-			argVal = fmt.Sprint(argVal)
+
+		argVal = fmt.Sprint(argVal)
```

Comment:
This will break other filters which aren't string based.

Reply:
I think I didn't get this. Previously we always assign the `argVal` interface with a string returned from fmt.Sprint() when the argName isn't "syscall". So I assumed that is safe to assign it always.

Reply:
@NDStrahilevitz let me know if it still have to be changed.

Reply:
This logic is contingent on the fact that we default to string filters for all arguments. In both cases, it's not entirely "correct", but the defaulting also isn't.